catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/TransformersCommand.java

```diff
@@ -0,0 +1,111 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.commands.catalog;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.karaf.shell.api.action.Command;
+import org.apache.karaf.shell.api.action.lifecycle.Service;
+import org.osgi.framework.ServiceReference;
+
+import ddf.catalog.Constants;
+import ddf.catalog.transform.InputTransformer;
+import javafx.util.Pair;
+
+/**
+ * Provides information on available transformers
+ */
+@Service
+@Command(scope = CatalogCommands.NAMESPACE, name = "transformers", description = "Provides information on available transformers.")
+public class TransformersCommand extends CatalogCommands {
+
+    private static final String MIME_TYPE = "mime-type";
+    private static final String SCHEMA = "schema";
+
+    private static final String NOT_AVAILABLE = "N/A";
+    private static final String LINE = "------------------------";
+
+    HashMap<String, Pair<String, ArrayList<String>>> transformersProperties;
+    ArrayList<ServiceReference<InputTransformer>> serviceReferences;
+    
+    @Override
+    protected Object executeWithSubject() throws Exception {
+        transformersProperties = new HashMap<>();
+        serviceReferences = (ArrayList<ServiceReference<InputTransformer>>) bundleContext.getServiceReferences(InputTransformer.class, null);
+
+        for (ServiceReference ref : serviceReferences) {
+            String id = getTransformerProperty(ref, Constants.SERVICE_ID);
+
+            if (!id.equals(NOT_AVAILABLE)) {
+                String schema = getTransformerProperty(ref, SCHEMA);
+                ArrayList<String> mimeType = getTransformerMimeProperty(ref);
+
+                transformersProperties.put(id, new Pair<>(schema, mimeType));
+            }
+        }
+
+        int activeTransformers = transformersProperties.size();
+
+        if (activeTransformers == 0) {
+            console.printf("There are no active transformers%n%n");
+
+        } else {
+            console.printf("%nActive Transformers: %d%n%s%n%n", activeTransformers, LINE);
+
+            int entryCount = 1;
+            for (Map.Entry<String, Pair<String, ArrayList<String>>> t: transformersProperties.entrySet()) {
```

Comment:
❗️ Now looking at how you're using this collection, I'm leaning very strongly toward you creating a private static inner class to hold the id, schema, and mime types. This would allow you to expand it in the future if needed and would clarify your access here. Instead of calls like `t.getKey()` and `t.getValue().getValue()`, you could call `t.getId()` and `t.getMimeTypes()`.

✏️ In fact, you could let the static class have methods that encapsulate some or all of this String formatting logic and then really simplify your code here.